reference-implementation/unittests/mhlo.cpp

```diff
@@ -31,160 +31,393 @@ const float EPSILON = 5e-4;
+  }
+  {
+    Tensor2D<long, 2, 2> x{-2, -1, 0, 2};
+    Tensor2D<long, 2, 2> expected_result{2, 1, 0, 2};
+    Tensor2D<long, 2, 2> result = mhlo::abs(x);
 
+    EXPECT_THAT(result, Pointwise(Eq(), expected_result));
+  }
+  {
+    Tensor3D<int32_t, 2, 1, 2> x{-2, -1, 0, 2};
+    Tensor3D<int32_t, 2, 1, 2> expected_result{2, 1, 0, 2};
+    Tensor3D<int32_t, 2, 1, 2> result = mhlo::abs(x);
+
+    EXPECT_THAT(result, Pointwise(Eq(), expected_result));
+  }
+  {
+    Tensor4D<int, 2, 2, 1, 2> x{-2, -1, 0, 0, 3, -3, -2, 1};
+    Tensor4D<int, 2, 2, 1, 2> expected_result{2, 1, 0, 0, 3, 3, 2, 1};
+    Tensor4D<int, 2, 2, 1, 2> result = mhlo::abs(x);
+
+    EXPECT_THAT(result, Pointwise(Eq(), expected_result));
+  }
   // TODO:: Test complex to real.
 }
 
 TEST(mhlo, ceil) {
   EXPECT_EQ(1.0, mhlo::ceil(0.7));
 
-  Tensor0D<float> t0{0.7f};
-  Tensor1D<float, 2> t1{1.6f, 2.0f};
-  Tensor2D<double, 2, 2> t2{2.1, 1.6, 0.0, 2.0};
-  Tensor3D<double, 2, 1, 2> t3{2.1, 1.6, 0.0, 2.0};
-  Tensor4D<double, 1, 2, 1, 2> t4{2.1, 1.6, 0.0, 2.0};
-
-  EXPECT_THAT(mhlo::ceil(t0), Pointwise(Eq(), {1.0f}));
-  EXPECT_THAT(mhlo::ceil(t1), Pointwise(Eq(), {2.0f, 2.0f}));
-  EXPECT_THAT(mhlo::ceil(t2), Pointwise(Eq(), {3.0, 2.0, 0.0, 2.0}));
-  EXPECT_THAT(mhlo::ceil(t3), Pointwise(Eq(), {3.0, 2.0, 0.0, 2.0}));
-  EXPECT_THAT(mhlo::ceil(t4), Pointwise(Eq(), {3.0, 2.0, 0.0, 2.0}));
+  {
+    Tensor0D<float> x{0.7f};
+    Tensor0D<float> expected_result{1.0f};
+    Tensor0D<float> result = mhlo::ceil(x);
+
+    EXPECT_THAT(result, Pointwise(Eq(), expected_result));
+  }
+  {
+    Tensor1D<float, 2> x{1.6f, 2.0f};
+    Tensor1D<float, 2> expected_result{2.0f, 2.0f};
+    Tensor1D<float, 2> result = mhlo::ceil(x);
+
+    EXPECT_THAT(result, Pointwise(Eq(), expected_result));
+  }
+  {
+    Tensor2D<double, 2, 2> x{2.1, 1.6, 0.0, 2.0};
+    Tensor2D<double, 2, 2> expected_result{3.0, 2.0, 0.0, 2.0};
+    Tensor2D<double, 2, 2> result = mhlo::ceil(x);
+
+    EXPECT_THAT(result, Pointwise(Eq(), expected_result));
+  }
+  {
+    Tensor3D<double, 2, 1, 2> x{2.1, 1.6, 0.0, 2.0};
+    Tensor3D<double, 2, 1, 2> expected_result{3.0, 2.0, 0.0, 2.0};
+    Tensor3D<double, 2, 1, 2> result = mhlo::ceil(x);
+
+    EXPECT_THAT(result, Pointwise(Eq(), expected_result));
+  }
+  {
+    Tensor4D<double, 1, 2, 1, 2> x{2.1, 1.6, 0.0, 2.0};
+    Tensor4D<double, 1, 2, 1, 2> expected_result{3.0, 2.0, 0.0, 2.0};
+    Tensor4D<double, 1, 2, 1, 2> result = mhlo::ceil(x);
+
+    EXPECT_THAT(result, Pointwise(Eq(), expected_result));
+  }
 }
 
 TEST(mhlo, convert) {
-  uint32_t a = 1;
-  uint64_t b = 1;
-  EXPECT_EQ(b, mhlo::convert<uint64_t>(a));
+  {
+    uint32_t x = 1;
+    uint64_t expected_result = 1;
+    uint64_t result = mhlo::convert<uint64_t>(x);
 
-  Tensor0D<uint32_t> t0{1};
-  auto lambda_0d = [&t0]() -> Tensor0D<size_t> {
-    return mhlo::convert<Tensor0D<size_t>>(t0);
-  };
-  EXPECT_THAT(lambda_0d(), Pointwise(Eq(), {1}));
+    EXPECT_EQ(result, expected_result);
+  }
 
```

Comment:
This is the only test with blank lines between the scopes. Would be neater to remove them